src/qt/editaddressdialog.cpp

```diff
@@ -164,3 +167,22 @@ void EditAddressDialog::setAddress(const QString &_address)
     this->address = _address;
     ui->addressEdit->setText(_address);
 }
+
+// Address changed
+void EditAddressDialog::addressEdited(const QString& text)
+{
+    if (text.isEmpty()) // Nothing entered
+    {
+        ui->errorMessage->setText("");
+        return;
+    }
+
+    if (!IsValidDestination(DecodeDestination(text.toStdString()))) // Invalid address
+    {
+        ui->errorMessage->setText(tr("Warning: Invalid Bitcoin address"));
+    }
+    else // Valid address
+    {
+        ui->errorMessage->setText("");
+    }
+}
```

Comment:
- clang-format
- missing newline at EOF
- can be simpified

```diff
@@ -171,18 +171,9 @@ void EditAddressDialog::setAddress(const QString &_address)
 // Address changed
 void EditAddressDialog::addressEdited(const QString& text)
 {
-    if (text.isEmpty()) // Nothing entered
-    {
-        ui->errorMessage->setText("");
-        return;
-    }
-
-    if (!IsValidDestination(DecodeDestination(text.toStdString()))) // Invalid address
-    {
+    if (!text.isEmpty() && !IsValidDestination(DecodeDestination(text.toStdString()))) {
         ui->errorMessage->setText(tr("Warning: Invalid Bitcoin address"));
-    }
-    else // Valid address
-    {
+    } else {
         ui->errorMessage->setText("");
     }
-}
\ No newline at end of file
+}
```

Reply:
Done. Thanks.